code/modules/reagents/Chemistry-Reagents/Drug-Reagents.dm

```diff
@@ -355,10 +352,95 @@
 			step(M, pick(cardinal))
 	M.Jitter(50)
 	M.Dizzy(50)
-	M.adjustToxLoss(5)
+	M.adjustToxLoss(4*REM)
 	M.adjustBrainLoss(10)
 	if(prob(50))
-		M.emote(pick("twitch","drool","moan"))
+		M.emote(pick("twitch","drool","moan","vomit","flip","scream"))
+	..()
+	return
+
+/datum/reagent/drug/heroin
+	name = "Heroin"
+	id = "heroin"
+	description = "An extremely advanced painkiller/narcotic. Heroin allows you to ignore all slowdown and grants you full immunity to stamina damage, but stuns are twice as effective against you. Mildly toxic."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = 0.75 * REAGENTS_METABOLISM
+	overdose_threshold = 15
+	addiction_threshold = 10
+	speedboost = IGNORE_SLOWDOWN
+
+/datum/reagent/drug/heroin/on_mob_life(mob/living/M)
+	M.setStaminaLoss(0)
+	if(M.stunned || M.weakened || M.paralysis)
```

Comment:
Is there any point for checking M.paralysis?
Since you're not extending the paralysis, all you're doing is adding an extra 0.5 per tick to their stuns which they'll be stunned anyway?